src/loaders.ts

```diff
@@ -101,6 +147,17 @@ export class EtherscanABILoader implements ABILoader {
                 compilerVersion: result.CompilerVersion,
                 runs: result.Runs,
 
+                getSources: () => {
+                    try {
+                        return Promise.resolve(this.#toContractSources(result));
+                    } catch (err: any) {
+                        throw new EtherscanABILoaderError("EtherscanABILoader getContract getSources error: " + err.message, {
+                            context: { url, address },
+                            cause: err,
+                        });
```

Comment:
```suggestion
                        return Promise.reject(new EtherscanABILoaderError("EtherscanABILoader getContract getSources error: " + err.message, {
                            context: { url, address },
                            cause: err,
                        }));
```
Need to Promise.reject or add async keyword or elase getSources().catch() will throw instead of catching

Reply:
Ah good point, thank you!!